cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java

```diff
@@ -692,11 +696,202 @@ private void handlePostSubscription(HttpServletRequest request, HttpServletRespo
                 response.getWriter().println("{\"success\":\"false\","
                             + "\"result\" : {" + orionJson.toJSONString() + "}");
             } // if else
+            
         } catch (Exception e) {
             LOGGER.error(e.getMessage());
         } // try catch
+        
     } // handlePostSubscription
     
+    private void handleDeleteSubscription(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("json;charset=utf-8");
+        
+        String subscriptionId = request.getParameter("subscription_id");
+        String ngsiVersion = request.getParameter("ngsi_version");
+        
+        if ((subscriptionId == null) || (subscriptionId.equals(""))) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            response.getWriter().println("{\"success\":\"false\","
+                    + "\"error\":\"Parse error, wrong parameter (subscription_id). Check it for errors.\"");
+            LOGGER.error("Parse error, wrong parameter (subscription_id). Check it for errors.\"}");
+            return;
+        } // if
+        
+        if ((ngsiVersion == null) || (ngsiVersion.equals(""))) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            response.getWriter().println("{\"success\":\"false\","
+                    + "\"error\":\"Parse error, wrong parameter (ngsi_version). Check it for errors.\"");
+            LOGGER.error("Parse error, wrong parameter (ngsi_version). Check it for errors.\"}");
+            return;
+        } // if
```

Comment:
I'm thinking that moving here the check about the version number would improve the method, because a lot of checks and other stuff is not executed if knowing in advance, at this point of the code, the version number is not correct.